### DISCUSSION TITLE
Request for Multi-App support in MeiliSearch

### DISCUSSION THREAD

**Discussion by Karman40** [meilisearch/product#606] [Feedback & Feature Proposal]
Dear MeiliSearch Team,

I hope this email finds you well. I am a big fan of MeiliSearch and have been using it as my primary search engine for several applications. However, I would like to request a future feature for MeiliSearch to support serving multiple, independent applications at once, similar to Algolia.

My idea would be as follows: I would focus on a single server to serve 5-10 clients, who would all have access only to their own index. Additionally, clients would have the option to create new indices. It might occur that all of the clients have an index with the same name, such as `users`.

This feature would greatly improve the flexibility and scalability of my projects, as I could easily manage and search different data sets in separate indices without having to switch between different search engines.

I believe this feature would greatly benefit the MeiliSearch community and make it a more competitive alternative to other search engines. I look forward to hearing your thoughts on this request.

Thank you for your time and consideration.

Best regards,
Zsombor KÁRMÁN

**Comment by macraig** [comment]
Hi again @Karman40 ! I'd love to learn more about your need for multi-app support. 

Is there a reason why you wouldn't want to spin up multiple Meilisearch instances? Is 5-10 clients an arbitrary number? If not, could you expand a bit more on this need?

Thanks!

**Reply by Karman40** [reply]
Hi @macraig.

If everyone got a Meilisearch instances, it would mean that I would have to do each update 5x, i.e. 5x the amount of work.

**Reply by macraig** [reply]
@Karman40 have you considered using [tenant tokens](https://www.meilisearch.com/docs/learn/security/tenant_tokens) to manage different clients? The caveat is that they won't be able to create indices on their own, is that a hard requirement?

We're also currently working on adding more features for team management to our [Cloud](https://blog.meilisearch.com/meilisearch-cloud-general-availability/) product. Although this wouldn't solve your use case completely today, it's the first step we're taking towards having a Role Management capability in the future. You can upvote [Team Management](https://roadmap.meilisearch.com/c/130-team-management) in our public roadmap to help us prioritize it.
